internal/host/statemachine.go

```
@@ -90,5 +91,32 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		PostTransition:   th.PostDisableHost,
 	})
 
+	// Enable host
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeEnableHost,
+		SourceStates: []stateswitch.State{
+			HostStatusDisabled,
+		},
+		DestinationState: HostStatusDiscovering,
+		PostTransition:   th.PostEnableHost,
+	})
+
+	// Enable disconnected host does not affect the state
```

If it does not affect the state, Maybe it should not be a transission?

---

How do we combine a state when sometimes it have a transition and sometimes we just allow it to run?
We can return an error in this state if it makes more sense?
I did the same trick with install https://github.com/filanov/bm-inventory/blob/master/internal/host/statemachine.go#L47

---

I'm open to suggestions.

---

Seems a bit to transition from a state to itself when it doesn't show up that way in the state machine diagram.  It make more sense to return 409.

---

Will do for this case.
What about install? there was a requirement to ignore disabled hosts. should it be a transition as i mentioned in the link?

---

Yes disabled->disabled makes sense for the install transition